From 669239820b449d8f137ef9974f117a1be313ea7c Mon Sep 17 00:00:00 2001
From: unknown author <cooker@mandrivalinux.org>
Date: Thu, 27 Sep 2007 15:22:23 +0000
Subject: [PATCH 4/6] xkb-fix_uz

---
 symbols/uz |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/symbols/uz b/symbols/uz
index b37a6ca..823ad71 100644
--- a/symbols/uz
+++ b/symbols/uz
@@ -27,7 +27,7 @@
 //
 // Latin variant is somewhat experimental, since there is no
 // official standard for it. It is a copy of "us" layout, where
-// apostrophe is replaced with U02BB.
+// backslash and pipe are replaced with U02BB and U02BC, respectively.
 
 default partial alphanumeric_keys
 xkb_symbols "cyrillic" {
@@ -91,5 +91,5 @@
 xkb_symbols "latin" {
     include "us"
     name[Group1]= "Uzbek (Latin)";
-    key <AC11>  { [0x010002BB, quotedbl] };
+    key <BKSL> { [ 0x010002BB, 0x010002BC ] };
 };
-- 
1.6.3.2
