From b75cdc6a6ce114984ccc5fa1c900c69979986ea8 Mon Sep 17 00:00:00 2001
From: unknown author <cooker@mandrivalinux.org>
Date: Thu, 26 Feb 2009 14:53:25 +0000
Subject: [PATCH 2/6] fi-kotoistus_classic_nbsp

---
 symbols/fi |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/symbols/fi b/symbols/fi
index ce9a366..66bee1c 100644
--- ./symbols/fi.orig	2010-05-13 22:07:22.000000000 +0200
+++ ./symbols/fi	2010-07-25 10:52:08.000000000 +0200
@@ -7,6 +7,14 @@
 //
 
 default  partial alphanumeric_keys
+xkb_symbols "kotoistus_classic_nbsp" {
+    include "fi(kotoistus)"
+    include "nbsp(level4)"
+
+    name[Group1]="Finland - Kotoistus with classic non-breaking space";
+};
+
+partial alphanumeric_keys
 xkb_symbols "kotoistus" {
 
     // Official keymap for Finland based on SFS 5966 standard
