diff -Naur xkeyboard-config-2.11/po/ru.po xkeyboard-config-2.11.tpg/po/ru.po
--- xkeyboard-config-2.11/po/ru.po	2014-01-15 15:32:02.000000000 +0000
+++ xkeyboard-config-2.11.tpg/po/ru.po	2014-02-06 16:54:56.039000042 +0000
@@ -132,6 +132,10 @@
 msgid "BTC 5090"
 msgstr "BTC 5090"
 
+#: ../rules/base.xml.in.h:40-1
+msgid "Altai"
+msgstr "Алтайская"
+
 #: ../rules/base.xml.in.h:28
 msgid "BTC 9019U"
 msgstr "BTC 9019U"
diff -Naur xkeyboard-config-2.11/rules/base.xml.in xkeyboard-config-2.11.tpg/rules/base.xml.in
--- xkeyboard-config-2.11/rules/base.xml.in	2014-01-15 01:02:39.000000000 +0000
+++ xkeyboard-config-2.11.tpg/rules/base.xml.in	2014-02-06 16:56:45.068993082 +0000
@@ -4240,6 +4240,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>alt</name>
+            <_description>Altai</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>typewriter-legacy</name>
             <_description>Russian (typewriter, legacy)</_description>
           </configItem>
diff -Naur xkeyboard-config-2.11/rules/evdev.xml.in xkeyboard-config-2.11.tpg/rules/evdev.xml.in
--- xkeyboard-config-2.11/rules/evdev.xml.in	2014-01-15 01:05:04.000000000 +0000
+++ xkeyboard-config-2.11.tpg/rules/evdev.xml.in	2014-02-06 16:57:34.277338909 +0000
@@ -4273,6 +4273,15 @@
         </variant>
         <variant>
           <configItem>
+            <name>alt</name>
+            <_description>Altai</_description>
+            <languageList>
+              <iso639Id>alt</iso639Id>
+            </languageList>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>cv</name>
             <_description>Chuvash</_description>
             <languageList>
diff -Naur xkeyboard-config-2.11/symbols/ru xkeyboard-config-2.11.tpg/symbols/ru
--- xkeyboard-config-2.11/symbols/ru	2013-12-18 22:33:58.000000000 +0000
+++ xkeyboard-config-2.11.tpg/symbols/ru	2014-02-06 17:02:26.474342917 +0000
@@ -265,6 +265,56 @@
 };
 
 partial alphanumeric_keys
+xkb_symbols "alt" {
+    include "ru(winkeys)"
+
+    name[Group1]= "Russia - Altai";
+
+    key.type[group1]="FOUR_LEVEL";
+
+    key <AE04> {   [ U0493,   U0492,      4,    U003B ] };
+    key <AE05> {   [ U04A5,   U04A4,      5,    U0025 ] };
+    key <AE06> {   [ U0458,   U0408,      6,    U003A ] };
+    key <AE07> {   [ U04F1,   U04F0,      7,    U003F ] };
+    key <AE08> {   [ U04E7,   U04E6,      8,    U002A ] };
+
+    key <AD01> {   [ Cyrillic_shorti,   Cyrillic_SHORTI,   U0473,   U0472 ] };
+    key <AD02> {   [ Cyrillic_tse,      Cyrillic_TSE,      U045E,   U040E ] };
+    key <AD03> {   [ Cyrillic_u,        Cyrillic_U,        U04F1,   U04F0 ] };
+    key <AD04> {   [ Cyrillic_ka,       Cyrillic_ka,       U049B,   U049A ] };
+    key <AD05> {   [ Cyrillic_ie,       Cyrillic_IE,       U0451,   U0401 ] };
+    key <AD06> {   [ Cyrillic_en,       Cyrillic_EN,       U04A5,   U04A4 ] };
+    key <AD07> {   [ Cyrillic_ghe,      Cyrillic_GHE,      U0493,   U0492 ] };
+    key <AD08> {   [ Cyrillic_sha,      Cyrillic_SHA,      U04A3,   U04A2 ] };
+    key <AD09> {   [ Cyrillic_shcha,    Cyrillic_SHCHA,    U04C8,   U04C7 ] };
+    key <AD10> {   [ Cyrillic_ze,       Cyrillic_ZE,       U04D9,   U04D8 ] };
+    key <AD11> {   [ Cyrillic_ha,       Cyrillic_HA,       U04BB,   U04BA ] };
+    key <AD12> {   [ Cyrillic_hardsign, Cyrillic_HARDSIGN, U04AF,   U04AE ] };
+    key <AC01> {   [ Cyrillic_ef,       Cyrillic_EF,       U04E1,   U04E0 ] };
+    key <AC02> {   [ Cyrillic_yeru,     Cyrillic_YERU,     U04F9,   U04F8 ] };
+    key <AC03> {   [ Cyrillic_ve,       Cyrillic_VE,       U0475,   U0474 ] };
+    key <AC04> {   [ Cyrillic_a,        Cyrillic_A,        U04D3,   U04D2 ] };
+    key <AC05> {   [ Cyrillic_pe,       Cyrillic_PE,       U04C4,   U04C3 ] };
+    key <AC06> {   [ Cyrillic_er,       Cyrillic_ER,       U04E3,   U04E2 ] };
+    key <AC07> {   [ Cyrillic_o,        Cyrillic_O,        U04E7,   U04E6 ] };
+    key <AC08> {   [ Cyrillic_el,       Cyrillic_EL,       U0513,   U0512 ] };
+    key <AC09> {   [ Cyrillic_de,       Cyrillic_DE,       U0458,   U0408 ] };
+    key <AC10> {   [ Cyrillic_zhe,      Cyrillic_ZHE,      U0497,   U0496 ] };
+    key <AC11> {   [ Cyrillic_e,        Cyrillic_E,        U04E9,   U04E8 ] };
+    key <AB01> {   [ Cyrillic_ya,       Cyrillic_YA,       U04B3,   U04B2 ] };
+    key <AB02> {   [ Cyrillic_che,      Cyrillic_CHE,      U04CC,   U04CB ] };
+    key <AB03> {   [ Cyrillic_es,       Cyrillic_ES,       U04AB,   U04AA ] };
+    key <AB04> {   [ Cyrillic_em,       Cyrillic_EM,       U0511,   U0510 ] };
+    key <AB05> {   [ Cyrillic_i,        Cyrillic_I,        U0456,   U0406 ] };
+    key <AB06> {   [ Cyrillic_te,       Cyrillic_TE,       U04EB,   U04EA ] };
+    key <AB07> {   [ Cyrillic_softsign, Cyrillic_SOFTSIGN, U0463,   U0462 ] };
+    key <AB08> {   [ Cyrillic_be,       Cyrillic_BE,       U0495,   U0494 ] };
+    key <AB09> {   [ Cyrillic_yu,       Cyrillic_YU,       U04B1,   U04B0 ] };
+
+    include "level3(ralt_switch)"
+};
+
+partial alphanumeric_keys
 xkb_symbols "cv" {
     include "ru(winkeys)"
 
@@ -344,14 +394,42 @@
     name[Group1]= "Komi";
 
     key.type[group1]="FOUR_LEVEL_ALPHABETIC";
-
+    key <AD01> { [ Cyrillic_shorti,   Cyrillic_SHORTI,   U04E5, U04E4 ] };
+    key <AD02> { [ Ukrainian_i,       Ukrainian_I,       U0446, U0426 ] }; //Cyr_tse
+    key <AD03> { [ Cyrillic_u,        Cyrillic_U,        U04F1, U04F0 ] };
+    key <AD04> { [ Cyrillic_ka,       Cyrillic_KA,       U049B, U049A ] };
+    key <AD05> { [ Cyrillic_ie,       Cyrillic_IE,       U0451, U0401 ] };
+    key <AD06> { [ Cyrillic_en,       Cyrillic_EN,       U050B, U050A ] };
+    key <AD07> { [ Cyrillic_ghe,      Cyrillic_GHE,      U0493, U0492 ] };
+    key <AD08> { [ Cyrillic_sha,      Cyrillic_SHA,      U04A5, U04A4 ] };
+    key <AD09> { [ U04E7,             U04E6,             U0449, U0429 ] }; //Cyr_shcha
+    key <AD10> { [ Cyrillic_ze,       Cyrillic_ZE,       U0505, U0504 ] };
+    key <AD11> { [ Cyrillic_ha,       Cyrillic_HA,       U0507, U0506 ] };
+    key <AD12> { [ Cyrillic_hardsign, Cyrillic_HARDSIGN, U0458, U0408 ] };
+    key <AC01> { [ Cyrillic_ef,       Cyrillic_EF,       U0515, U0514 ] };
+    key <AC02> { [ Cyrillic_yeru,     Cyrillic_YERU,     U04F9, U04F8 ] };
+    key <AC03> { [ Cyrillic_ve,       Cyrillic_VE,       U04EB, U04EA ] };
+    key <AC04> { [ Cyrillic_a,        Cyrillic_A,        U04D3, U04D2 ] };
+    key <AC05> { [ Cyrillic_pe,       Cyrillic_PE,       U0503, U0502 ] };
+    key <AC06> { [ Cyrillic_er,       Cyrillic_ER,       U0517, U0516 ] };
 // cyrilllic o with diaeresis 
     key	<AC07> { [ 	Cyrillic_o,		Cyrillic_O,		
 			U04E7,  		U04E6		] };
-
+    key <AC08> { [ Cyrillic_el,       Cyrillic_EL,       U0509, U0508 ] };
+    key <AC09> { [ Cyrillic_de,       Cyrillic_DE,       U0501, U0500 ] };
+    key <AC10> { [ Cyrillic_zhe,      Cyrillic_ZHE,      U0497, U0496 ] };
+    key <AC11> { [ Cyrillic_e,        Cyrillic_E,        U04E9, U04E8 ] };
+    key <AB01> { [ Cyrillic_ya,       Cyrillic_YA,       U0519, U0518 ] };
+    key <AB02> { [ Cyrillic_che,      Cyrillic_CHE,      U04F5, U04F4 ] };
+    key <AB03> { [ Cyrillic_es,       Cyrillic_ES,       U050D, U050C ] };
+    key <AB04> { [ Cyrillic_em,       Cyrillic_EM,       U04CE, U04CD ] };
 // hard I as additional key for soft Cyrillic_I 
     key <AB05> { [      Cyrillic_i,		Cyrillic_I,
 			Ukrainian_i,     	Ukrainian_I     ] };
+    key <AB06> { [ Cyrillic_te,       Cyrillic_TE,       U050F, U050E ] };
+    key <AB07> { [ Cyrillic_softsign, Cyrillic_SOFTSIGN, U0463, U0462 ] };
+    key <AB08> { [ Cyrillic_be,       Cyrillic_BE,       U04DF, U04DE ] };
+    key <AB09> { [ Cyrillic_yu,       Cyrillic_YU,       U04DD, U04DC ] };
 
     include "level3(ralt_switch)"
 };
@@ -510,18 +588,26 @@
 // http://www.marlamuter.ru/
 partial alphanumeric_keys
 xkb_symbols "chm" {
-    include "ru(common)"
+    include "ru(winkeys)"
 
     name[Group1]= "Mari";
 
     key.type[group1]="FOUR_LEVEL_ALPHABETIC";
-    key <AC04> { [      Cyrillic_a,     Cyrillic_A,    U04D3,       U04D2     ] };
-    key <AD03> { [      Cyrillic_u, Cyrillic_U,    U04F1,       U04F0     ] };
-    key <AC02> { [      Cyrillic_ui,    Cyrillic_UI,   U04F9,       U04F8     ] };
-    key <AD06> { [      Cyrillic_n, Cyrillic_N,    U04A5,       U04A4     ] };
-    key <AC11> { [      Cyrillic_e, Cyrillic_E,    U04E9,       U04E8     ] };
-    key <AC10> { [      Cyrillic_zh,    Cyrillic_ZH,   U04EB,       U04EA     ] };
-    key <AC01> { [      Cyrillic_f, Cyrillic_F,     UF537,      UF536     ] };
+    key <AC04> { [ Cyrillic_a,      Cyrillic_A,      U04D3,       U04D2     ] };
+    key <AD03> { [ Cyrillic_u,      Cyrillic_U,      U04F1,       U04F0     ] };
+    key <AC02> { [ Cyrillic_ui,     Cyrillic_UI,     U04F9,       U04F8     ] };
+    key <AD06> { [ Cyrillic_n,      Cyrillic_N,      U04A5,       U04A4     ] };
+    key <AC11> { [ Cyrillic_e,      Cyrillic_E,      U04E9,       U04E8     ] };
+    key <AC10> { [ Cyrillic_zh,     Cyrillic_ZH,     U04EB,       U04EA     ] };
+//    key <AC01> { [ Cyrillic_f,      Cyrillic_F,      UF537,       UF536     ] };
+    key        <AC07> { [ Cyrillic_o,      Cyrillic_O,      U04E7,       U04E6     ] };
+    key        <AD01> { [ Cyrillic_shorti, Cyrillic_SHORTI, U04E5,       U04E4     ] };
+    key <AB02> { [ Cyrillic_che,    Cyrillic_CHE,    U04F5,       U04F4     ] };
+    key <AB05> { [ Cyrillic_i,      Cyrillic_I,      U0456,       U0406     ] };
+    key <AD10> { [ Cyrillic_ze,     Cyrillic_ZE,     U04DF,       U04DE     ] };
+    key <AD05> { [ Cyrillic_ie,     Cyrillic_IE,     U0451,       U0401     ] };
+    key <AC09> { [ Cyrillic_de,     Cyrillic_DE ,    U0458,       U0408     ] };
+    key <AD07> { [ Cyrillic_ghe,    Cyrillic_GHE,    U0493,       U0492     ] };
 
     include "level3(ralt_switch)"
 };
