--- xkeyboard-config-2.11.tpg/symbols/ch	2013-12-18 22:33:58.000000000 +0000
+++ xkeyboard-config-2.11.tpg/symbols/ch.tpg	2014-02-06 17:06:14.294591989 +0000
@@ -75,6 +75,17 @@
 };
 
 partial alphanumeric_keys
+xkb_symbols "de_scriptdeadkeys" {
+    // modify the basic Swiss German layout to avoid dead keys with symbols
+    // commonly used in scripting (^, ', `, ...), but to keep the otherwise
+    // useless ¨ key a dead key
+    include "ch(basic)"
+    name[Group1]= "Switzerland - German, some dead keys";
+    key <AE11> { [  apostrophe, question,        acute  ] };
+    key <AE12> { [ asciicircum,    grave,   asciitilde  ] };
+};
+
+partial alphanumeric_keys
 xkb_symbols "fr" {
 
     include "ch(basic)"
